yt/frontends/stream/data_structures.py

```diff
@@ -653,18 +677,12 @@ def load_uniform_grid(data, domain_dimensions, length_unit=None, bbox=None,
     domain_left_edge = np.array(bbox[:, 0], 'float64')
     domain_right_edge = np.array(bbox[:, 1], 'float64')
     grid_levels = np.zeros(nprocs, dtype='int32').reshape((nprocs,1))
-    number_of_particles = data.pop("number_of_particles", 0)
-    # First we fix our field names
-    field_units, data = unitify_data(data)
-
-    dshape = tuple(domain_dimensions)
-    for field_name in data:
-        fshape = data[field_name].shape
-        if len(fshape) == 3 and fshape != dshape:
-            msg = ("Input data shape %s for field %s does not match provided "
-                   "domain_dimensions %s!")
-            msg = msg % (fshape, field_name, dshape)
-            raise RuntimeError(msg)
+    # If someone included this throw it away--old API
+    data.pop("number_of_particles", 0)
```

Comment:
It should work correctly--I ran the tests with `number_of_particles` included first before changing them and they all passed. I will add a warning here.